src/sliding-sync-sdk.ts

```diff
@@ -406,9 +406,51 @@ export class SlidingSyncSdk {
         // this helps large account to speed up faster
         // room::decryptCriticalEvent is in charge of decrypting all the events
         // required for a client to function properly
-        const timelineEvents = mapEvents(this.client, room.roomId, roomData.timeline, false);
+        let timelineEvents = mapEvents(this.client, room.roomId, roomData.timeline, false);
         const ephemeralEvents = []; // TODO this.mapSyncEventsFormat(joinObj.ephemeral);
 
+        // TODO: handle threaded / beacon events
+
+        if (roomData.initial) {
+            // we should not know about any of these timeline entries if this is a genuinely new room.
+            // If we do, then we've effectively done scrollback (e.g requesting timeline_limit: 1 for
+            // this room, then timeline_limit: 50).
+            const knownEvents: Record<string, boolean> = {};
+            room.getLiveTimeline().getEvents().forEach((e) => {
+                knownEvents[e.getId()] = true;
+            });
```

Comment:
Should this not be a Set?

Reply:
Ah yes, too much Go... (which has no `Set` type!!!!)

Reply:
You have Generics now, can now finally make a generic Set! :D